backend/courses/util.py

```diff
@@ -672,3 +674,352 @@ def get_semesters(semesters=None, verbose=False):
+
+    def find_lectures_on_day(lectures, day):
+        """
+        Finds all the lectures on a given day.
+
+        Parameters:
+            lectures (list): A list of lectures.
+            day (str): The day to search for lectures on.
+
+        Returns:
+            list: A list of tuples containing the start time, end time,
+            course code, and section ID for each lecture on the given day.
+        """
+        lectures_on_day = []
+        for lecture in lectures:
+            for section in lecture:
+                for meeting in section["meetings"]:
+                    if meeting["day"] == day:
+                        # Adds a randomizer to randomize the order they are fed to the scheduler
+                        lectures_on_day.append(
+                            (
+                                float(meeting["start"]),
+                                float(meeting["end"]) + 0.001 * random.random(),
+                                "-".join(section["id"].split("-")[0:2]),
+                                section["id"],
+                                section["credits"],
+                            )
+                        )
+        return lectures_on_day
+
+    def schedule_to_section(schedules):
+        """
+        Given a list of schedules,
+        returns a list of sections by stripping the time intervals from the schedule list.
+
+        Parameters:
+            schedules (list): A list of schedules.
+
+        Returns:
+            list: A list of sections.
+        """
+        # Removes the time intervals from the schedule list
+        sections = []
+        for s in schedules:
+            sections.append((s[3], s[4]))
+        return sections
+
+    def remove_duplicates(given_l):
+        """
+        Removes duplicates from a list.
+        Example:
+        remove_duplicates([1,1,1,3,5,7,7]) = [1,3,5,7]
+        """
+        newl = []
+        [newl.append(x) for x in given_l if x not in newl]
+        return newl
+
+    def choose_class_hash(given_l):
+        """
+        Given a list of section nodes, chooses one section per class from each schedule randomly.
+
+        Parameters:
+            l (list): A list of section nodes.
+
+        Returns:
+            list: A list of chosen sections, one per class.
+        """
+        hash = {}
+        courses = []
+        for node in given_l:
+            class_name = "-".join(node[0].split("-")[0:2])
+            if class_name not in hash.keys():
+                hash[class_name] = [node]
+            else:
+                hash[class_name].append(node)
+        for key in hash.keys():
+            # Chooses a random lecture for each class
+            courses.append(random.choice(hash[key]))
+        return courses
+
+    def credit_count(sections):
+        acc = 0
+        for section in sections:
+            acc += section[1]
+        return acc
+
+    def check_if_schedule_possible(schedule, courses, unwanted_intervals):
```

Comment:
Is the logic too complicated to incorporate this directly until whatever interval solver problem you have? Trying to determine the real need for a separate function.

Reply:
There might be a better way to do this, but the use of it is to resolve conflicts after merging all the days together. It also makes adding recitations a little cleaner, as its kinda like "induction": if schedule works, does schedule + recs work?